File: src/main/java/com/chatchatabc/jpademojava/infra/DataInit.java
package com.chatchatabc.jpademojava.infra;

import com.chatchatabc.jpademojava.domain.service.RoleService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

@Component
public class DataInit implements CommandLineRunner {
    @Autowired
    private RoleService roleService;

    /**
     * Create roles on runtime
     *
     * @param args
     * @throws Exception
     */
    @Transactional
    @Override
    public void run(String... args) throws Exception {
        roleService.createRoles();
    }
}

File: src/main/kotlin/com/chatchatabc/jpademo/infra/DataInit.kt
package com.chatchatabc.jpademo.infra

import com.chatchatabc.jpademo.domain.service.RoleService
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
class DataInit(
    private val roleService: RoleService
) : CommandLineRunner {

    /**
     * Create Roles on runtime
     */
    @Transactional
    override fun run(vararg args: String?) {
        roleService.createRoles()
    }
}